README.md

```diff
@@ -2,44 +2,145 @@
+Reading the grib weather data requires the eccodes library: `apt-get install libeccodes-dev`
+
+### Installing python libraries
 `pip3 install -r requirements.txt`
 
 Verify you can run the unit tests `python -m unittest`
 
-If running in a container, to start a notebook use `jupyter notebook --NotebookApp.ip=0.0.0.0`
+To start jupyter notebook run `jupyter notebook --NotebookApp.ip=0.0.0.0`
+Now you can open the example notebook and execute the cells in the demonstration.
+
+<!-- Phase 2 (after FTR accepted): add docker container in July -->
 
 ## Usage
-Models can be composed of mixins for various estimators and forecast processes.
+Models can be composed of mixins for various estimators and forecast processes. These composable
+pieces can be put together in different ways to solve many problems. The RegularTimeSeriesModel is the base.
+The core that problem specific parts are added to when forecasting or gap filling a particular timeseries.
+The estimator is the next essential building block. It can be a Classifier or a Regressor. There are many 
+different numerical techniques that can be applied via the estimator. The process is the last essential 
+component. It defines the timeseries being forecast and the available feature data that might have predictive value.
+Having composed a Model class from these three parts, it is then up to the user to create an instance of the class that 
+with configuration arguments that tune the model features for the specific meter load or pv forecast. 
+
 
-An ordinary least squares model for a Balancing Area Hourly Load
+<!-- Explain what this does - for lawyers... point to ipynb example -->
+<!-- Consider adding sklearn model diagram if it would help (rather than hurt) the explanation -->
+
+### Compose a model class
+
+A model using an Ordinary Least Squares estimator to forecast a Balancing Area Hourly Load
 ```python
 class NpOlsModel(BalancingAreaHourly, LinearRegressor, RegularTimeSeriesModel):
   pass
 ```
-An XGBoost model for Balancing Area Hourly Load
+
+A model using an XgBoost estimator to forecast AMI (meter) Hourly Load
 ```python
-class XgbModel(BalancingAreaHourly, XgbRegressor, RegularTimeSeriesModel):
+class XgbModel(AmiHourlyForecast, XgbRegressor, RegularTimeSeriesModel):
   pass
 ```
 
-The initialization arguments control feature construction.
+Additional behaviors like filters can also be added via composition. See the notebook demo for examples.
 
+### Configure a model instance
+The initialization arguments control feature construction. The configuration arguments are specific to the
+process and the estimator used to compose the model.
 
-### Trained models should be fully serializable using Cloud Pickle for maximum flexibility
+As an example, we can configure XgbModel from above with three types of features: lagged correlations, one hot encoded 
+day-of-week values and a harmonic feature that decomposes time into a sine and cosine waves with the specified frequencies.
```

Comment:
Hmmm, I like that harmonic feature is defined here. Maybe also define lagged correlation and one hot encoded day-of-week. Maybe put these in a bulleted list.

P3